controller/transfer/util.py

```diff
@@ -98,7 +99,15 @@ def convert_to_record_dict(
         os.remove(file_name)
     run_limit_checks(df, project_id, user_id)
     run_checks(df, project_id, user_id)
-    return df.to_dict(orient="records")
+    return check_and_convert_category_for_unknown(df)
+
+
+def check_and_convert_category_for_unknown(df: pd.DataFrame) -> Dict:
+    df_check = pd.DataFrame(df.to_dict(orient="records"))
```

Comment:
why is this copied? we let the garbage collector tidy up the after this anyway so no need to copy it. also if we need it we could use the provided copy methods and not converted it to a dict and back